### ISSUE TITLE
Rule Change: semi-spacing for leading semicolon

### ISSUE THREAD

**Issue by ackvf** [eslint/eslint#15752]
### What rule do you want to change?

semi-spacing

### What change to do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
// expected
console.log('hi')
;(something ??= {_default: 0})[key] = 1

// wrong
console.log('hi')
; (something ??= {_default: 0})[key] = 1

// super wrong
console.log('hi')
;        (something ??= {_default: 0})[key] = 1
```

### What does the rule currently do for this code?

Nothing, it keeps both _expected_ and _wrong_ intact in their original form.

### What will the rule do after it's changed?

Both cases look as the _expected_.

- notifies of extra space after leading semi
- autofix to remove the extra space

### Participation

- [ ] I am willing to submit a pull request to implement this change.

## semi-spacing
https://eslint.org/docs/rules/semi-spacing

### Rule Details
This rule doesn't check spacing in the following cases:

- The spacing after the semicolon if it is the first token in the line.

In this request, I specifically want to disallow space between a leading semi and the following statement.

My suggestion would be to add one new option with the following `default` value,

```
{"afterFirst": false}
```

Implementing `beforeFirst` is not necessary as it is handled by `indent`.

This change is driven by an undesired auto formatting issue with VSCode's default JS/TS language features formatter
https://github.com/microsoft/vscode/issues/146620 Disable inserting spaces around leading semicolon

**Comment by mdjermanovic** [comment]
Hi @ackvf, thanks for the issue!

Per our rules policies [stylistic rules are frozen](https://github.com/eslint/eslint#stylistic-rule-updates), so we are not adding new options to the `semi-spacing` rule. You can always copy the existing rule into a custom rule and make it behave the way you want.
